sdk/core/core-lro/review/core-lro.api.md

```diff
@@ -59,16 +87,62 @@ export interface LroResponse<T = unknown> {
     rawResponse: RawResponse;
 }
 
+declare namespace Next {
```

Comment:
hmm... a new pattern

Reply:
Yeah, as [design discussion](https://microsoft.sharepoint.com/:fl:/s/61554b56-c90b-4961-ba7c-61caae235335/EW431kB80rlOpXmFqAJWEscBZjN_Gfy6hnDr_jkiLj9cIg?e=zWchnd&nav=cz0lMkZzaXRlcyUyRjYxNTU0YjU2LWM5MGItNDk2MS1iYTdjLTYxY2FhZTIzNTMzNSZkPWIhWkhjMnhhbl9nVXVNc3RET0NHaTNKMjZEQ09nVk5FRkxtTDg4UUNsZUpwaklYajhzTWlFSFRZVC02bTRGV2JkNSZmPTAxREdGMk9FM09HN0xFQTdHU1hGSEtLNk1GVkFCRk1FV0gmYz0lMkYmZmx1aWQ9MSZhPUxvb3BBcHAmcD0lNDBmbHVpZHglMkZsb29wLXBhZ2UtY29udGFpbmVyJng9JTdCJTIydyUyMiUzQSUyMlQwUlRVSHh0YVdOeWIzTnZablF1YzJoaGNtVndiMmx1ZEM1amIyMThZaUZhU0dNeWVHRnVYMmRWZFUxemRFUlBRMGRwTTBveU5rUkRUMmRXVGtWR1RHMU1PRGhSUTJ4bFNuQnFTVmhxT0hOTmFVVklWRmxVTFRadE5FWlhZbVExZkRBeFJFZEdNazlGTkZSRlRWRTBOMUpHUVRkYVJUSlBSazVEUkRSV1dEWTBNa1UlM0QlMjIlMkMlMjJpJTIyJTNBJTIyODM0YTY3ZjktMTVlNy00NDkxLWIwMjAtYzg5YTljMGYxODZhJTIyJTdE) if we deliever in subpath then the cjs customers may not refer this and it would be a concern for us.

Reply:
I guess another option we didn't discuss and may be reasonable is:

1. Create a beta branch with this new work replacing the existing implementation
2. Release beta versions with the new API (v3-beta)
3. Once we are ready to GA the new API release with a major version bump
4. Have a flag in the generator to generate for v2 but default to v3. And autorest to default to v2
5. We can release hotfixes to v2 if needed

This way we don't maintain 2 parallel implementations within the same package, while we allow for a phased migration to the new API for existing libraries that don't want to take the breaking change right away.

Reply:
I am okay with this option considering the new lro would be in preview at first and once stable we could have confidence to accept this breaking in core.

@bterlson @deyaaeldeen @xirzec @qiaozha any concerns?